scrapy/utils/deprecate.py

```diff
@@ -121,3 +122,13 @@ def _clspath(cls, forced=None):
     if forced is not None:
         return forced
     return '{}.{}'.format(cls.__module__, cls.__name__)
+
+
+def update_classpath(path):
+    """Update a deprecated path from an object with its new location.
+
+    This function tries to load the object based on the given path, hence it'll
+    only work if the old path is still supported.
+    """
+    obj = load_object(path)
```

Comment:
As the list of relocated modules is known, my vote is to add a function to each component (extensions, spidermw, downmw) so each of them rewrite classpaths using string only operations.